### ISSUE TITLE
Custom Resources pages using wrong API version for Tekton

### ISSUE THREAD

**Issue by jstrachan** [lensapp/lens#3592]
I tried to view PipelineRuns using a recent [tekton install](https://tekton.dev/) and it didn't show any.

It did show TaskRun resources but when trying to edit them it converted them to the old API version so you can't use any of the new schema when editing:

```yaml 
apiVersion: tekton.dev/v1alpha1
kind: TaskRun
...
```

rather than their most recent version which shows up in kubectl / VS Code k8s extension:

```yaml 
apiVersion: tekton.dev/v1beta1
kind: TaskRun...
```

I guess the bug is that the CRD extension is not using the latest API version when browsing/editing the CRDs? Both PipelineRun and TaskRun have 2 versions and it should use the 2nd (latest) version by default.

e.g. see the versions:
<img width="614" alt="pipeline-run-crd" src="https://user-images.githubusercontent.com/30140/128996865-b6c5e563-86d7-4f0e-83d7-83f8ad67cc83.png">

**Comment by SomaticIT** [comment]
I have a similar issue with `cert-manager`.
`cert-manager` upgraded several times their CRD, the definition contains 4 versions: (`v1alpha2`, `v1alpha3`, `v1beta1`, `v1`).

If I inspect the definition, I see that only v1 is actually served and stored (`served: true, storage: true`).

When `Lens` tries to fetch these resources, it uses only the first version available (here: `v1alpha2`) but this version is not served anymore.

I see two ways to fix this bug:
1. If we could not make multiple requests, the simplest solution would be to fetch only the latest version (not the first one).
2. If we are able to do multiple requests, we should fetch all versions that are both `served` and `stored`.

*Note: the bad version is also appearing on `Definitions` tab. This should also be fixed.*

**Comment by rulim34** [comment]
I think this issues is simillar with https://github.com/kubernetes/dashboard/issues/6632.

**Comment by SomaticIT** [comment]
You're right, this is a similar issue.

Both `lens` and `k8s dashboard` are not taking in account the `served` flag when they retrieve crd resources.
This means, it should be fixed here and in `k8s dashboard` (out of the scope of this issue).

**Comment by Nokel81** [comment]
I think this was fixed by https://github.com/lensapp/lens/pull/4553. Is this still a problem in 5.5?

**Comment by SomaticIT** [comment]
Thank you for your response.
You're right, it seems to be fixed in v5.5.

I think you can close this issue.
